### PR TITLE
Updating material conversions to include all items with "legacy" as the item format

### PR DESCRIPTION
**Pull request by ShadowShift**
All numbers were obtained through a script that ran through legacy values in 

https://api.wynncraft.dev/v3/item/database?fullResult

### CODE REVIEW

File: Data-Storage/material_conversions.json [hunk 1]
```diff
@@ -1,187 +1,1057 @@
...
+	},
+	{
+		"id": 29,
+		"type": 0,
+		"name": "sticky_piston"
+	},
+	{
+		"id": 30,
+		"type": 0,
+		"name": "cobweb:
```
Comment (kristofbolyai) [inline Data-Storage/material_conversions.json:105]: error

File: Data-Storage/material_conversions.json [hunk 2]
```diff
@@ -1,187 +1,1057 @@
...
+	},
+	{
+		"id": 168,
+		"type": 1,
+		"name": "prismarine_bricks"
+	},
+	{
+		"id": 168,
+		"type": 2,
+		"name": "dark_prismarine
```
Comment (kristofbolyai) [inline Data-Storage/material_conversions.json:395]: error

File: Data-Storage/material_conversions.json [hunk 3]
```diff
@@ -1,187 +1,1057 @@
...
+	},
+	{
+		"id": 321,
+		"type": 0,
+		"name": "painting"
+	},
+	{
+		"id": 323,
+		"type": 0,
+		"name": "sign:
```
Comment (kristofbolyai) [inline Data-Storage/material_conversions.json:675]: error

**Review by kristofbolyai — COMMENTED** [review]
I suggest you run this json through some kind of validator.

**Comment by ShadowShift** [comment]
will do

**Comment by magicus** [comment]
Hm? These are vanilla mappings; they should not be affected by any changes on Wynncraft's side.

Where did you get the mappings from?

**Comment by magicus** [comment]
Also, could we get rid of all `"type": 0`, and let it be default? Or will Wynntils barf at that?

**Comment by kristofbolyai** [comment]
> Hm? These are vanilla mappings; they should not be affected by any changes on Wynncraft's side.
> 
> Where did you get the mappings from?

It is not affected. This PR, despite it's slightly bad naming, extends our current material conversion list by everything that is currently missing for the 2.1 legacy item codes. That is because we couldn't find a great 1.12 id -> 1.20 mapping JSON before, so we agreed on adding these edge-cases by hand. Shadow wants to extend this list.

**Comment by kristofbolyai** [comment]
> Also, could we get rid of all `"type": 0`, and let it be default? Or will Wynntils barf at that?

No, we absolutely cannot change this JSON's format, it's mission critical for most item types.

**Comment by magicus** [comment]
> That is because we couldn't find a great 1.12 id -> 1.20 mapping JSON before, so we agreed on adding these edge-cases by hand. Shadow wants to extend this list.

I see. So this should really be replaced by a proper vanilla mapping file. But in the meantime this will have to do, I guess.

**Comment by magicus** [comment]
> No, we absolutely cannot change this JSON's format, it's mission critical for most item types.

I was just hoping that we had read this "the right way" using GSON; then int fields should be initialized to 0 if they are missing from the json. It would keep the json file a bit simpler, since most materials has just a single type. But it's not important.

**Comment by ShadowShift** [comment]
> Hm? These are vanilla mappings; they should not be affected by any changes on Wynncraft's side.
> 
> Where did you get the mappings from?

I wrote a script to go through the api and flag all ingredients with legacy formatting. Just remembered that there were 2 items with it as well (Gert Relik and Wybel Paw) and added the missing ids from those. I attached a screenshot of the script and its output. Items arent included from there as I had forgotten about them at the time.

image

[message(7).txt](https://github.com/user-attachments/files/16805639/message.7.txt)

**Comment by ShadowShift** [comment]
Done. Switched from 1 tab to 2 spaces

File: Data-Storage/material_conversions.json [hunk 4]
```diff
@@ -102,13 +917,18 @@
   {
     "id": 383,
     "type": 0,
-    "name": "wolf_spawn_egg"
+    "name": "spawn_egg"
```
Comment (kristofbolyai) [inline Data-Storage/material_conversions.json:920]: Should you have changed this?

Comment (ShadowShift) [reply Data-Storage/material_conversions.json]: Oh nope thats mb

**Review by magicus — APPROVED** [review]

**Comment by kristofbolyai** [comment]
Thanks

**Pull request merged**